cayenne/src/main/java/org/apache/cayenne/access/translator/select/GroupByStage.java

```diff
@@ -41,7 +41,7 @@ public void perform(TranslatorContext context) {
         }
     }
 
-    private boolean hasAggregate(TranslatorContext context) {
+    static boolean hasAggregate(TranslatorContext context) {
```

Comment:
This again breaks stage logic separation, but I don't see a good option here as with the `hasToManyJoins()` method

Reply:
A possibility is moving it to TranslatorContext, what do you think ?

Reply:
Moved shared logic into StageUtil instead.